### PR TITLE
Getting started with machine learning in Matlab

### LINKED ISSUE #2604: Getting started with machine learning in Matlab
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Getting started with machine learning
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction:
Machine learning is a type of computer science that allows computer programs to learn and improve on their own. In the past, computers could only do what they are programmed to do but with machine learning, they could behave like human being and gain knowledge based on their past experiences. let's say for example we want a program that that can tell the difference between apples and bananas. First we give the computer some label pictures of each. The program will identify the pattern in the fruit and begin to remember them. It can then use these memories to identify the unlabeled and tell which is which.
Machine learning is encountered in daily basis for example, in the medical industry, it is applied for predicting life span, organizing patients data e.t.c. Matlab provides a machine learning toolbox that makes machine learning easier. In this article, we will look at the basis of machine learning.
### Key takeaways: 
- Overview of machine learning.
- Algorithms available in Matlab.
- Overcoming machine learning challenges using Matlab.
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by peterAdongo**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      This closes #2604.
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
@peterAdongo 

Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Thank you.

**Comment by ahmadmardeni1** [comment]
this fixes #2604

**Comment by srishilesh** [comment]
@peterAdongo 

Great article. On a quick glance, I see that this is a vast topic, which I believe could not be covered in a single article. I would suggest to break this down, and write articles focusing on specific topic.

Moreover, I see majority of your article was intro to AI - Supervised, unsupervised learning, and its types. We have already covered this, in the other articles.

I feel it would be better to rename the title and dive deep into that topic specifically. Just make sure that you don't have conflicts with already [existing articles](https://www.section.io/engineering-education/search/?q=matlab) in MATLAB. And, reference these articles in yours, as you see fit.

Ping me once you have decided on the new topic.

Thanks.

**Comment by linetachieng** [comment]
@srishilesh I change the topic to ' how to implement linear regression in Matlab' is that ok??

**Comment by srishilesh** [comment]
@linetachieng 

Sounds good.

But, changing only the title wouldn't be enough. You will have to dig deeper into linear regression concepts. (We already have articles explaining the concepts, so I would suggest to choose some other type of regression).

Fyi, there are 12 types of regression. You could choose something else which you don't find here. You'll have to explain the concepts in-detail, and go through the code tutorial step-by-step. Make sure you choose a well defined problem for the code tutorial.

Thanks.

cc. @hectorkambow

**Comment by hectorkambow** [comment]
@lalith1403 may be able to add some insight on this type of topic as well.

File: content/articles/getting-started-with-machine-learning-using-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+Matlab does not only provides a strong environment for iterative exploration, but it also has apps that allow you to explore the impacts of different models on your data. The framework makes it easy to iterate over different algorithms and measure their accuracy.
+One can also incorporate a parallel computing paradigm to speed up computations. Once you are happy with the models, You can integrate them into larger data and analytics and decision making workflow.
+
+### Overview 0f machine learning
+Machine learning is classified into two branches, that is;
+- Supervised learning.
+- Unsupervised learning.
+
+### Unsupervised learning
+Here, you group your data based on some similarities or characteristics. It is not a must you know the grouping. This grouping can be done using clustering techniques. Clustering is the grouping of the dataset based on its characteristics. Matlab provides a variety of clustering algorithms. The algorithm is later grouped into partitioning algorithms and overlapping algorithms.
```
Comment (lalith1403) [inline content/articles/getting-started-with-machine-learning-using-matlab/index.md:29]: Clustering is just one form of unsupervised learning algorithm

File: content/articles/getting-started-with-machine-learning-using-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
```
Comment (lalith1403) [inline content/articles/getting-started-with-machine-learning-using-matlab/index.md:30]: The distinction between partition and overlapping algorithms is not clear. More explanation is required.

File: content/articles/getting-started-with-machine-learning-using-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+### i) Classification
+This technique is used for discrete response. For example, if you want to predict tumours size into small, medium or large, then it is a classification problem and uses classification techniques to solve this. The algorithm for classification is such as support vector machines, discriminant analysis, naive Bayes and nearest neighbour.
+
+### ii) Regression
+This technique is for the continuous response. For example is if you are trying to predict electricity demand from a grid in KW/hr, then it is a regression problem and requires q regression techniques. The algorithm for regression is such as neural network, decision tree, ensemble method, non-linear regression and linear regression.
+
+### Workflow for supervised learning
+Here, you first import data, preprocess the data. This may include creating some visualization and identifying variables of interest. In the next step, you select your model and train it on the data and measure its accuracy. Once you have selected the best model, you can use it to predict the new input data set.
```
Comment (lalith1403) [inline content/articles/getting-started-with-machine-learning-using-matlab/index.md:47]: Test the model on test set. That needs to be mentioned.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Several points to be noted in this article. When we write an article, what is the case we are trying to make? What narrative do we control and present to the reader? These questions need to be answered. In your case, MATLAB as an alternative to Python is what I understand. This is something you should focus on. Explaining supervised learning, unsupervised learning is not something we would touch in such an article as it wouldn't help us make the case for MATLAB as an alternative. I hope that makes sense. Would love to hear your thoughts on this.

**Comment by lalith1403** [comment]
As mentioned earlier, pick a well defined problem/ statement that you want to write on. That way, we can focus on the core content we want to put out. We generally avoid elementary topics such as linear regression in MATLAB/Python unless it gives us a better explanation than the ones that exist.

**Comment by peterAdongo** [comment]
@srishilesh  ok let me talk about how to solve the Fourier series using Matlab

cc @hectorkambow

**Comment by peterAdongo** [comment]
@srishilesh ok let me talk about how to solve the Fourier series using Matlab

cc @hectorkambow, @ahmadmardeni1

**Comment by srishilesh** [comment]
@peterAdongo 

I find one such documentation by MathWorks [here](https://in.mathworks.com/help/curvefit/fourier.html). How exactly will yours be different from this? What topics do you wish to cover under this?

I don't have much idea about Fourier series. Probably, @lalith1403 could help us here.

Thanks. 

cc. @hectorkambow

**Comment by peterAdongo** [comment]
@srishilesh here I cover deeply and a more concise article. I will talk about how to find Fourier coefficients, Fourier series sine and cosine form. I will also look at the complex-valued coefficients and Fourier series complex-exponential form which is not in the link you provided

cc @hectorkambow

**Comment by srishilesh** [comment]
@peterAdongo 

Let's wait for @lalith1403 's opinion.

**Comment by lalith1403** [comment]
Are you suggesting writing a new article? If yes, can you go through the topic suggestion form again? I will need more information on the article and the concepts that you will be covering. Fourier Series has been covered well over the net. Scope exists in the form of unique projects that you can work on. If you can visualize the complex coefficients and implement them on custom image and music data, that would be an excellent project. What do you think? 

If you are trying to make modifications to this article, can you give me more information on how you want to link ML and Fourier series?

**Comment by hectorkambow** [comment]
Who is the author here @peterAdongo  or @linetachieng

**Comment by peterAdongo** [comment]
@hectorkambow the author is @peterAdongo

**Pull request closed without merging**